src/dc_receive_imf.rs

```diff
@@ -480,16 +461,14 @@ fn add_parts(
         // the mail is on the IMAP server, probably it is also delivered.
         // We cannot recreate other states (read, error).
         state = MessageState::OutDelivered;
-        *from_id = DC_CONTACT_ID_SELF;
```

Comment:
this already happens before add_parts() is called at all, and from_id is passed in.